### ISSUE TITLE
sec: Possible Malicious non-public fork with misleading doc website

### ISSUE THREAD

**Issue by JakobLichterfeld** [teslamate-org/teslamate#3828]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

In #3825 a user reported an issue, but was not using any official teslamate images.

The fork in non-public but there seems to be a misleading website as well, they removed `.org` from official https://docs.teslamate.org/  and replace with`.com.cn`

### Relevant log output

```shell
n.a
```

### Type of installation

Docker

### Version

all

**Comment by JakobLichterfeld** [comment]
@brianmay any ideas how we can handle this? Report the website?

**Comment by brianmay** [comment]
We have a MIT license, where the only requirement is that the license is not removed.

Disclaimer: I am not a lawyer.

I think this does mean people can make private forks and claim that they are official Teslamate. Which is unfortunate from the support point of view.

This is a case where GPL would help, any changes that are distributed must be distributed in source too. But maybe too late for Teslamate.

I think the best we can do is what you did, say we can only support the official Teslamate from the official Teslamate github repo. People are free to make forks, but we can't support them. Any problems need to be reproduced with the offical image before we can support them.

I note that there image has 15 docker layers:

https://hub.docker.com/layers/dhuar/teslamate/1.28.5.1/images/sha256-925e9a9b1a6808f437e8932d3aae514bca06cee2d6761e7e920af8bb609a8c32?context=explore

Ours only has 12:

https://hub.docker.com/layers/teslamate/teslamate/1.28.5/images/sha256-8c5ac2651ff4177e9bc90d5fff8c4f6a30acb27292437b90f600f5eb61671c7a?context=explore

My brain is a bit asleep right now, not sure if this is significant or not. Will come back to this tomorrow. It wouldn't surprise me if they are missing recent changes in Teslamate or something like that.

**Comment by brianmay** [comment]
The typical tool to solve the issue of "people can distribute malicious code and call it my project" is to register a trademark on the project name. But you would need a non-zero budget to do so. You probably would need some sort of legal entity to to hold the trademark too, and need to do it in such a way that it can't be taken over in the future and used for a closed proprietary solution.

**Comment by JakobLichterfeld** [comment]
Thanks for your thoughts!

I also agree on the trademark side, thus you can see often claims on official websites of open-source projects, that this is the only official site and that there are scam versions of it.

It just makes me sad to see how open source is being abused and people are being potentially harmed.

**Comment by brianmay** [comment]
There might be open source lobby groups that will do the trademark stuff for us. e.g. I think the FSF does that, and there are others.

I just saw this article, but haven't read it: https://google.github.io/opencasebook/trademarks/

**Comment by kkthxbye-code** [comment]
Here's the image diff:

```
-----File-----

These entries have been added to dhuar/teslamate:1.28.5.1:
FILE                                                                                               SIZE
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app-7263e078e657fc31aeb2ac0b9af10d40.js           556.4K
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app-7263e078e657fc31aeb2ac0b9af10d40.js.gz        146.7K

These entries have been deleted from dhuar/teslamate:1.28.5.1:
FILE                                                                                               SIZE
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app-a826aecba29ce34b45932ec26cadd2f2.js           556.4K
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app-a826aecba29ce34b45932ec26cadd2f2.js.gz        146.7K

These entries have been changed between teslamate/teslamate:1.28.5 and dhuar/teslamate:1.28.5.1:
FILE                                                                                  SIZE1         SIZE2
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app.js                               556.4K        556.4K
/opt/app/lib/teslamate-1.28.5/priv/static/assets/app.js.gz                            146.7K        146.7K
/opt/app/lib/ex_cldr-2.37.5/ebin/Elixir.Cldr.Validity.U.beam                          45.9K         45.9K
/opt/app/lib/ex_cldr-2.37.5/ebin/Elixir.Cldr.Config.beam                              39.9K         40K
/opt/app/lib/phoenix_live_view-0.18.18/ebin/Elixir.Phoenix.Component.beam             34.1K         34.1K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.SettingsLive.Index.beam        18.9K         18.9K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.LayoutView.beam                14.2K         14.4K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.Router.beam                    11.5K         11.5K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.ImportLive.Index.beam          11K           11K
/opt/app/lib/tzdata-1.1.1/ebin/Elixir.Tzdata.Util.beam                                9K            9K
/opt/app/lib/teslamate-1.28.5/priv/static/cache_manifest.json                         7K            7K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMate.Import.LineParser.beam            6.7K          6.7K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.GeoFenceLive.Index.beam        6.1K          6.1K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMate.Locations.Geocoder.beam           5.1K          5.1K
/var/cache/ldconfig/aux-cache                                                         4.9K          4.9K
/opt/app/lib/tzdata-1.1.1/ebin/Elixir.Tzdata.Parser.beam                              4.8K          4.8K
/opt/app/lib/ex_cldr-2.37.5/ebin/Elixir.Cldr.Locale.Loader.beam                       4.6K          4.6K
/var/log/dpkg.log                                                                     4.4K          4.4K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMateWeb.CarLive.Index.beam             4K            4K
/var/log/apt/term.log                                                                 3K            3K
/opt/app/lib/teslamate-1.28.5/ebin/Elixir.TeslaMate.HTTP.beam                         1.3K          1.3K
/var/log/apt/history.log                                                              517B          517B
/etc/shadow                                                                           496B          496Bb
/opt/app/releases/COOKIE                                                              56B           56B
```

Seems likely that this is the source repo: https://github.com/mywind2020/teslamate-cn

Seems like he's not pushing his changes anymore though. While there could be malicious stuff hidden, I think it's also likely that it's just a bootleg Chinese version. 

The user created an issue in this repo a couple of days ago, so you might be able to reach out.

https://github.com/teslamate-org/teslamate/issues/3830

**Comment by kkthxbye-code** [comment]
@JakobLichterfeld - I apologize for trying to give some context. I have deleted it. You seem emotionally involved as well, prehaps not the best look for a maintainer.

**Comment by brianmay** [comment]
After what happened with xz (maintainer facing mental issues was bullied until he handed over the project to a malicious developer), it is quite clear you can't be too careful here. We don't know anything about this project or its maintainer or if they are trustworthy or not. I think we need to err on the side of caution here. Particularly as this code has access to Tesla Tokens and - I believe - has significant use across the globe.

There is nothing wrong with creating a fork of TeslaMate, that is fine. But we cannot check every fork and rate it for trustworthiness either.

There is nothing wrong with posting a diff of the two images either. I am a bit surprised in got deleted. However, as the xz experience shows us malicious actors can be patient, waiting years while they develop good changes before they switch to malicious changes. If we endorsed the image now, could will stand by this in, say 2 years time?

In this case I don't even see a good reason for a fork, if there is something lacking with the official images, then we would be more then willing to review any pull requests to fix this. Forks of open source software are generally only required if upstream is moving in a way that contributors don't like or is failing to merge pull requests for some reason.

**Comment by JakobLichterfeld** [comment]
I fully agree with that.  

> I am a bit surprised in got deleted

I also don't understand why kkthxbye deleted his comment.

**Comment by kkthxbye-code** [comment]
@JakobLichterfeld - Your answer was rude and weirdly emotional. I was in no way promoting the image, just trying to have calmer heads prevail and provide some actual investigation. Your goal seems different so I deleted it. Have a good one.

**Comment by oivindoh** [comment]
A cursory search yields https://github.com/mywind2020/teslamate-cn/commits/master/ as the likely source of those images

**Comment by vanling** [comment]
Looking at their (translated) description:
> Chinese version of TeslaMate, fixed the problem of unavailability of map and location in China, added battery health display.

and commits https://github.com/teslamate-org/teslamate/compare/master...mywind2020:teslamate-cn:master

it does not look very "malicious" but you never know these days.

**Comment by haerr** [comment]
https://docs.teslamate.com.cn/docs/teslamatebox

This website engages in commercial activities by adding translated documents to the website and embedding new pages in the documents to sell products.

**Comment by linhongjun** [comment]
I'm using this one in China on my VPS. It's easy to deploy on server with only one script to run.
It fixed the problem of map and location in China, no need to access to github.com blocked by GFW.
They sell a ready-to-use TeslaMate Box at the price of 40 USD which can store data home.

**Comment by zjxuzhj** [comment]
It ranks first on Baidu, China's search engine, for the keyword "TeslaMate." This is highly misleading for those who are just starting to explore TeslaMate, and it is primarily aimed at promoting their own hardware, which is sold for $40. Such practices are truly distasteful.

**Comment by linhongjun** [comment]
> It ranks first on Baidu, China's search engine, for the keyword "TeslaMate." This is highly misleading for those who are just starting to explore TeslaMate, and it is primarily aimed at promoting their own hardware, which is sold for $40. Such practices are truly distasteful.

Github is banned in China, Normal user won't be able to access official TeslaMate and there's a existing problem that map can't be displayed in China. You can use this version without his hardware, What's your problem? What did you do to TeslaMate? What did you do to the community and TeslaMate Chinese users?

**Comment by brianmay** [comment]
I see several issues with this that some people might have:

* They are using open source software, contributed by volunteers in their spare time, and maybe profiteering(?) from it.
* Maybe they make the source code available. Assuming the links above are correct. But is there any assurance the built image actually corresponds to the github source? I personally haven't seen anything connecting the github to the docker hub image other this this thread.
* AFAIK they are not contributing changes back to the community.
* People could get confused and believe they were using an official release. This has happened already in #3825. This in turn could make us look bad if something goes wrong.
* Security fixes or other urgent updates might be delayed. Most likely because Tesla broke something again.

Apologies in advance if I got something wrong here.

Unfortunately https://github.com/teslamate-org/teslamate/compare/master...mywind2020:teslamate-cn:master seems to be full of changes we made. Which might be because github is getting confused with the fact our changes have been merged into their fork, and we are asking it to merge them back again. Would have hoped github could cope better with this actually.

Looking under the "Files Changed" tab it appears the bulk of the changes are Chinese translations to the grafana dashboards. I don't think we have any support for localization for the grafana dashboards. I can understand why these changes have not been contributed back. But if I am right, we could use a PR to fix this. 

The minor change to the documentation should perhaps be fixed in our code.

Nothing here that could cause the problems describe in #3825. It is likely it was an old version relative to our master, and that may have caused some confusion.

Despite having concerns, what they are doing is perfectly legal and allowed by the MIT license.

I don't think there is any more we can do here for this issue. Except maybe:

* Add some information to the documentation that there are unofficial forks with the same name, we are not associated with them in anyway, and as a result we cannot vouch for them. The only official sites are https://github.com/teslamate-org/teslamate/, https://hub.docker.com/r/teslamate/teslamate and https://docs.teslamate.org/.
* Is there any way we can contact them and ask them nicely to clarify that their project is a fork?
* Somehow add ability to do translations of the grafana dashboard - assuming this is possible. Maybe we need to preprocess the files or something. https://github.com/teslamate-org/teslamate/discussions/1610

**Comment by brianmay** [comment]
Watching this video reminds me of this ticket. https://www.youtube.com/watch?v=dJJvq3dpylM

It shows what can happen if somebody makes changes to an upstream project but doesn't make it obvious that their distribution isn't the official upstream distribution.

**Comment by madswaord** [comment]
> In [#3825](https://github.com/teslamate-org/teslamate/issues/3825) a user reported an issue, but was not using any official teslamate images.
> 
> The fork in non-public (which can be an indication of malicious activity) but there seems to be a misleading website as well, they removed `.org` from official https://docs.teslamate.org/ and replace with`.com.cn`
> 
> The issue is not the translation of the doc (even if this must result in an out-of date doc as it is not maintained) the installation instructions point to the non-public fork.
> 
> TeslaMate officially does have full Chinese translation and use Chinese API endpoint when needed.

may i ask where to find full chinese translation, the datas in the grafana still in english.

![Image](https://github.com/user-attachments/assets/a34c23db-1408-4486-bd13-f99f7d857b6c)

**Comment by JakobLichterfeld** [comment]
> may i ask where to find full chinese translation, the datas in the grafana still in english.

Grafana does not support any localisation, so there is no localisation for any language and we decided for english.
All current solutions (like duplicating dashboards and the translate) result in heavy maintenance effort.

TeslaMate itself is translated in many different languages.

**Comment by carrot-eggs** [comment]
https://www.teslamate.com.cn/
The site was shut down due to a warning from Tesla’s Justice Department, and the issue could be shut down.

**Comment by brianmay** [comment]
Seems concerning. I am not aware of them doing anything malicious or illegal. Do you know why?

**Comment by carrot-eggs** [comment]
> Seems concerning. I am not aware of them doing anything malicious or illegal. [Do](url) you know why?

Please refer to the content of the image below.

![Image](https://github.com/user-attachments/assets/26a607fe-cacc-4101-b7d6-5f8dba368296)

<img width="478" height="655" alt="Image" src="https://github.com/user-attachments/assets/b3f10e78-5bf5-4040-b15a-63eaf61b503f" />

**Comment by carrot-eggs** [comment]
> Seems concerning. I am not aware of them doing anything malicious or illegal. Do you know why?

In the end, it's still because of TeslaMate's infrastructure, where ports 4000 and 3000 are not encrypted by default. The Chinese Cyberspace Administration believes there is a risk of data leakage.To avoid unnecessary risks, Tesla's legal department in China has taken all products based on TeslaMate offline and issued them a lawyer's letter.If TeslaMate had encrypted access for ports 4000 and 3000 by default, these second development TeslaMate products in China would not have been shut down.It is recommended to implement default encryption configuration for TeslaMate's web access.That's all.

**Comment by carrot-eggs** [comment]
https://www.teslamate.cloud/
This site is also closed!，Also based on the TeslaMate secondary development.

**Comment by brianmay-actions** [comment]
Translated using google translate, 1st image:

Risk warning regarding Tesla vehicle data exposure due to a vulnerability in the open-source tool TeslaMate

To the Cyberspace Administration of the Party Committees of all counties (cities, and districts), and relevant municipal units:

According to a report from the Guangdong Provincial Cybersecurity Situation Awareness Platform, the open-source tool TeslaMate was recently exposed to an unauthorized access vulnerability. This third-party open-source tool uses Tesla's official API to provide data analysis, monitoring, and statistical notifications to Tesla owners. Its default configuration opens port 4000 to the internet without authentication. Furthermore, its port 3000 runs a dashboard with a default or weak password, potentially exposing Tesla vehicles' real-time location, historical trajectory, and vehicle status data. TeslaMate has issued a security advisory and provided a hardening guide: https://docs.teslamate.org/docs/advanced_guides/traefik/.

All county (city, district) Party committee cyberspace administration offices and relevant municipal units are requested to guide cadres and staff in their respective regions and units to investigate the use of the TeslaMate tool and the impact of the vulnerability, and urge them to implement security reinforcements to eliminate network security risks.

**Comment by brianmay-actions** [comment]
2nd image:

Tesla's third-party tool Tesla Mate exposed security vulnerabilities

Can lead to the owner's real-time positioning driving habits exposure

Tesla's third-party data tool Tesla Mate was recently exposed to an unauthorized access vulnerability. Its default configuration opens port 4000 to the Internet without setting up authentication. At the same time, its port 3000 runs a dashboard with a default password or a weak password, which may lead to the leakage of data such as the vehicle's real-time location, historical trajectory, and vehicle status.

TeslaMate is a third-party open-source tool that uses Tesla's official API to provide Tesla owners with data analysis, monitoring, statistics, and notifications. TeslaMate has released security advisories and provides security enhancements. TeslaMate users should investigate its usage and vulnerability impact, implement security enhancements, and mitigate network security risks.

**Comment by brianmay** [comment]
Random thoughts:

* Above seems to concerned more about specific installations, not the software.
* It is standard practice that websites will be non-encrypted and expect user to add some sort of front end that does the TLS.
* But people are installing Teslamate on public systems without knowing what they are doing. Even worse is when you sell access to poorly configured systems to other people (teslamate.cloud looks like it may have been in this category).
* We could create scripts, etc. to setup self signed certificates on initial default install, but security issues will remain (e.g. no authentication, man-in-middle attack, etc.).
* Maybe default config should listen to localhost only by default. Users' would get broken system by default, not a wide open system.

**Comment by carrot-eggs** [comment]
In order to avoid unnecessary trouble, in China, Tesla China banned the teslamate data of major cloud manufacturers (Tencent Cloud, AliCloud, Huawei Cloud) deployed in China, and it is possible to block the Owners API in China. It is recommended that Tesla actively switch the default interface from the Owners API to the Fleet API.

**Comment by JakobLichterfeld** [comment]
As expected, the potential malicious non-public fork has caused issues for all users.

**Comment by carrot-eggs** [comment]
<img width="848" height="796" alt="Image" src="https://github.com/user-attachments/assets/0f4a0682-4faa-4af0-8662-39939799946f" />

**Comment by carrot-eggs** [comment]
I received an email from Tesla China suggesting that I was logged into the Tesla account using a third-party application, and there was a risk of information leakage, because I used TeslaMate, and I used the version downloaded on GitHub (org version) that you did not use secondary development.
The translation of the message is as follows:

Security Tip: Third Party Applications

We note that you may be logged into a Tesla account using an insecure third-party application or service. Such third-party applications or services may result in your personal and vehicle information being stolen by others on the network, infringing on your personal interests and information security.
To protect your account and information security, we recommend that you:
(1) immediately modify your Tesla account password and set up multi-factor authentication to ensure account security;
(2) We recommend that you close or remove Tesla account authorization links on any unofficial platform to prevent the disclosure of data and your personal information;
Tesla has always put user privacy and data security first. If you are unsure how to do this, you can check how you create, update or delete a Tesla account or contact the customer support team, and we will provide you with further support.

Tesla China

**Comment by carrot-eggs** [comment]
Should I stop using TeslaMate immediately?

**Comment by carrot-eggs** [comment]
<img width="366" height="462" alt="Image" src="https://github.com/user-attachments/assets/e52b7e5f-77be-47b3-bd08-4ff313a26da3" />

The Tesla app sent me the same.

**Comment by WinterBoring** [comment]
我今天也收到邮件了！用的是dhuar/teslamate，刚才看文档主页已经显示关闭！中国车主应该怎么操作？？？

**Comment by haerr** [comment]
使用当前 github官方 teslamate服务进行部署，并且部署至本地，不要部署至阿里云等公有云

WinterBoring ***@***.***> 于2025年9月24日周三 14:44写道：

**Comment by JakobLichterfeld** [comment]
To enable a broader audience to follow, please only use the English language here.

Regarding the unrelated comments: never expose any service to the internet.
> If you intend to access TeslaMate from the Internet, the recommended way is to use a secure connection (such as a VPN, Cloudflare Tunnel, Tailscale, Zero Tier and a reverse proxy for portless access) for secured access to your TeslaMate instance outside your home network. Alternatively, you can use a reverse proxy (such as Traefik, Apache2 or Caddy) with appropriate hardening to secure your TeslaMate instance before expose it to the internet, check out the [advanced guides with Traefik](https://docs.teslamate.org/docs/advanced_guides/traefik) for an example how to use Traefik with TeslaMate. Or you can use the [advanced guides with Apache](https://docs.teslamate.org/docs/advanced_guides/apache) to set up TeslaMate with Apache2, TLS and HTTP Basic Auth.

**Comment by jaypark0006** [comment]
> 我今天也收到邮件了！用的是dhuar/teslamate，刚才看文档主页已经显示关闭！中国车主应该怎么操作？？？

@WinterBoring You should back up your teslamate database, stop it, and switch to the release version of this repository. 你应该将自己的teslamate备份数据库，然后停掉，换成本仓库的release版本。


> > 使用当前 github官方 teslamate服务进行部署，并且部署至本地，不要部署至阿里云等公有云
> > WinterBoring _**@**_.***> 于2025年9月24日周三 14:44写道：
> > […](#)
> 
> 我一直用的就是你说的这个方式部署的官方版本，我没有在公有云部署，我家也没公网IP，完全内网部署也收到了Tesla 中国的邮件提醒。

@carrot-eggs I guess this prompt means that as long as the token is exposed to a program other than the app, as long as there is a call, an email and message reminder will be sent to the car owner. It's just a reminder. 我猜测这个提示是只要暴露了token在非app之外的程序，只要有调用就会向车主发邮件和消息提醒。只是一个提醒而已

**Comment by JakobLichterfeld** [comment]
> To enable a broader audience to follow, please only use the English language here.

Every non english message will be deleted from now on.
